### ISSUE TITLE
"dotnet" exited with code 1

### ISSUE THREAD

**Issue by joaommvsoares** [AArnott/CodeGeneration.Roslyn#81]
Hello,

My sample works fine in version 0.3.9, but when I try with 0.4.49 it trows the error at compile time:

1>  No executable found matching command "dotnet-codegen"
1>C:\Trabalho\Porjs\GeneratedCode\packages\CodeGeneration.Roslyn.BuildTime.0.4.49\build\CodeGeneration.Roslyn.BuildTime.targets(20,5): error MSB6006: "dotnet" exited with code 1.

I even tried to include at project:
<DotNetCliToolReference Include="dotnet-codegen" Version="0.4.49" />

I used VS21017 to CodeGenerator.DLL and VS2015 to use it on GeneratedCode.exe (my final project must be in VS2015 for now…)

What seems to be the problem?

Thanks,
João Miguel

**Comment by joaommvsoares** [comment]
I tried that, but It doen't work.
It throws an error with tag version.

**Comment by amis92** [comment]
Are you using the SDK-style csproj?

The file should start with

```xml
<Project Sdk="Microsoft.NET.Sdk">
…
```

If you are, please attach the error you see. This DotNetCliToolReference is required.

**Comment by amis92** [comment]
Re-reading the original post: using the generator requires running `dotnet codegen` DotNetCliTool which is only available if the project that's supposed to run it has `DotNetCliToolReference`. In turn, that element is only supported in SDK-style `csproj` files, which in turn again are only supported in VS2017.

TLDR this project only works in VS2017+ all the way.
